src/main/java/com/sysdig/jenkins/plugins/sysdig/CLIDownloadAction.java

```diff
@@ -0,0 +1,260 @@
+package com.sysdig.jenkins.plugins.sysdig;
+
+import java.io.BufferedInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.net.URL;
+import java.nio.channels.Channels;
+import java.nio.channels.ReadableByteChannel;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.attribute.PosixFilePermissions;
+import java.util.Scanner;
+
+import hudson.model.Run;
+import jenkins.model.RunAction2;
+
+public class CLIDownloadAction implements RunAction2 {
+    private String name;
+    private String os;
+    private String arch;
+    private String version;
+    private transient Run run;
+    private String url;
+    private String basePath = "/var/jenkins_home";
+
+    private Path apath;
+
+    public String chwd() {
+
+        return this.apath.toAbsolutePath().toString();
+    }
+
+    private void makeUrl() {
+        try {
+            @SuppressWarnings("deprecation")
+            URL uarl = new URL("https", "download.sysdig.com", "/scanning/sysdig-cli-scanner/latest_version.txt");
+            Scanner s = new Scanner(uarl.openStream());
+            if (this.version.isEmpty() || this.version.equalsIgnoreCase("latest")){
+                this.version = s.nextLine();
+            }
+            this.url = String.format(
+                    "https://download.sysdig.com/scanning/bin/sysdig-cli-scanner/%s/%s/%s/sysdig-cli-scanner",
+                    this.version,
+                    this.os.toLowerCase(), this.arch.toLowerCase());
+            s.close();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+     private void downloadUsingNIO( String file) throws IOException {
+        URL url = new URL(this.url);
```

Comment:
```suggestion
        URL url = new URI.create(this.url).toURL();
```

But even better, can we change the type of `this.url` to be of type `URL`, so we don't need to re-compute it over and over?

In the constructor we could do:
```
this.url = URI.create(
    String.format(
        "https://download.sysdig.com/scanning/bin/sysdig-cli-scanner/%s/%s/%s/sysdig-cli-scanner",
        this.version,
        this.os.toLowerCase(), 
        this.arch.toLowerCase()
    )
).toURL();
```